ofono/plugins/ril.c

```diff
@@ -475,6 +479,42 @@ void ril_switchUser()
 
 }
 
+/* return: Null-terminated path string. Ownership transferred. */
+static char *ril_socket_path()
+{
+	GError *err = NULL;
+	GKeyFile *keyfile = NULL;
+	char *res = NULL;
+
+	keyfile = g_key_file_new();
+	g_key_file_set_list_separator(keyfile, ',');
+
+	if (!g_key_file_load_from_file(keyfile, RILMODEM_CONF_FILE, 0, &err)) {
+		g_key_file_free(keyfile);
```

Comment:
glib doesn't like deleting null so glib code tends to produce this kind of client code. Nevertheless I too prefer one point of cleanup so changed it that way.